app/pages/TaxYearSelectionPage.scala

```diff
@@ -16,12 +16,35 @@
 
 package pages
 
-import models.TaxYearSelection
+import models.{PSub, TaxYearSelection, UserAnswers}
+import models.TaxYearSelection._
 import play.api.libs.json.JsPath
 
+import scala.util.{Success, Try}
+
 case object TaxYearSelectionPage extends QuestionPage[Seq[TaxYearSelection]] {
 
   override def path: JsPath = JsPath \ toString
 
   override def toString: String = "taxYearSelection"
+
+  override def cleanup(value: Option[Seq[TaxYearSelection]], userAnswers: UserAnswers): Try[UserAnswers] =
+    value match {
+      case Some(taxYears) =>
+        import models.PSubsByYear.formats
+
+        userAnswers.get(SummarySubscriptionsPage) match {
+          case Some(psubsByYear) =>
+            val newPsubsByYear: Map[Int, Seq[PSub]] = taxYears.map {
+              year =>
+                getTaxYear(year) -> psubsByYear.getOrElse(getTaxYear(year), Seq.empty)
+            }.toMap
+
+            userAnswers.set(SummarySubscriptionsPage, newPsubsByYear)
+          case _ =>
+            Success(userAnswers)
+        }
+      case _ =>
+        Success(userAnswers)
+    }
```

Comment:
Refactor to this maybe? :+1: 

    (value, userAnswers.get(SummarySubscriptionsPage)) match {
      case (Some(taxYearSelection), Some(psubsByYear)) =>
        val newPsubsByYear: Map[Int, Seq[PSub]] = taxYearSelection.map {
          year =>
            getTaxYear(year) -> psubsByYear.getOrElse(getTaxYear(year), Seq.empty)
        }.toMap

        userAnswers.set(SummarySubscriptionsPage, newPsubsByYear)
      case _ =>
        Success(userAnswers)
    }